Apply comments from the parent PR (#2929)
FuelLabs/fuel-core: .changes/breaking/2887.md

Bump Rust version to `1.85.0`.

> Starting this release, newly generated WASM state transition function is not more compatible with old versions of the `fuel-core`. So, the change breaks forward compatibility for the network and each node should start to use a new `fuel-core` release.

Reviewer: nit:
```suggestion
Starting with this release, newly generated WASM state transition functions are no longer compatible with old versions of the `fuel-core`. So, the change breaks forward compatibility for the network and each node should start to use a new `fuel-core` release.
```
